app/services/movie_service.rb

```diff
@@ -63,4 +52,39 @@ def movie_info_genres(genres)
       genre[:name]
     end
   end
+
+  def movie_info_cast(api_movie_id)
+    cast_info = get_data((url_storage(movie_id: api_movie_id)[:movie_cast]))
+    movie_cast = {}
+    cast_info[:cast].each_with_index do |cast_member, index|
+      movie_cast[(cast_member[:name])] = cast_member[:character] if index < 10
+    end
+    movie_cast
+  end
+
+  def movie_info_reviews(api_movie_id)
+    review_info = get_data((url_storage(movie_id: api_movie_id)[:movie_reviews]))
+    movie_reviews_info = { total_reviews: review_info[:total_results] }
+    results_page_count((url_storage(movie_id: api_movie_id)[:movie_reviews])).times do |n|
+      review_info = get_data((url_storage(movie_id: api_movie_id, num: n)[:movie_reviews]))
+      review_info[:results].each do |review|
+        movie_reviews_info[(review[:author_details][:username])] = review[:content]
+      end
+    end
+    movie_reviews_info
+  end
+
+  def url_storage(movie_id: 1, num: 0, query: '', api_key: ENV['API_KEY'])
```

Comment:
This is great that we are filtering out our api key so that it does not get exposed, however, I think we might actually have it backwards here. The generic key, `"<TMDB_API_KEY>"` is what we want to use as a placeholder instead of our real api_key that we are currently making accessible across the HTTP requests.